lib/logstash/inputs/rabbitmq.rb

```diff
@@ -310,6 +312,28 @@ def get_properties(metadata)
           acc
         end
       end
+
+      METADATA_NONE     = Set[].freeze
+      METADATA_BASIC    = Set[:headers,:properties].freeze
+      METADATA_EXTENDED = Set[:headers,:properties,:payload].freeze
+      METADATA_DEPRECATION_MAP = { 'true' => 'basic', 'false' => 'none' }
+
+      private
+      def extract_metadata_level(metadata_enabled_setting)
```

Comment:
This code is good, I would cover with a couple of unit tests

Reply:
I've added extensive unit tests to rspec suite